### PR TITLE
[PyTorch] Add BERT FineTuning

### PR DESCRIPTION
**Pull request by AnirudhDagar**
*Description of changes:*

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

File: chapter_computer-vision/image-augmentation.md [hunk 1]
```diff
@@ -262,7 +262,12 @@ def train_batch_ch13(net, features, labels, loss, trainer, devices,
 #@tab pytorch
 #@save
 def train_batch_ch13(net, X, y, loss, trainer, devices):
-    X, y = X.to(devices[0]), y.to(devices[0])
+    if isinstance(X, list):
+        # Required for BERT Finetuning (Sec 15.7)
+        X = [inp.to(devices[0])for inp in X]
```
Comment (astonzhang) [inline chapter_computer-vision/image-augmentation.md:267]: space before `for`
`inp` is not clear
Required for BERT Fine-tuning (to be covered later)

File: chapter_convolutional-neural-networks/lenet.md [hunk 2]
```diff
@@ -256,7 +256,12 @@ def evaluate_accuracy_gpu(net, data_iter, device=None): #@save
     # No. of correct predictions, no. of predictions
     metric = d2l.Accumulator(2)
     for X, y in data_iter:
-        X, y = X.to(device), y.to(device)
+        if isinstance(X, list):
```
Comment (astonzhang) [inline chapter_convolutional-neural-networks/lenet.md:259]: same here

File: chapter_computer-vision/image-augmentation.md (hunk 1 above)
Comment (astonzhang) [reply chapter_computer-vision/image-augmentation.md:267]: maybe `inp` -> 'x'?

File: chapter_natural-language-processing-applications/natural-language-inference-bert.md [hunk 3]
```diff
@@ -71,11 +89,34 @@ def load_pretrained_model(pretrained_model, num_hiddens, ffn_num_hiddens,
...
+def load_pretrained_model(pretrained_model, num_hiddens, ffn_num_hiddens,
+                          num_heads, num_layers, dropout, max_len, devices):
+    data_dir = d2l.download_extract(pretrained_model)
+    # Define an empty vocabulary to load the predefined vocabulary
+    vocab = d2l.Vocab()
+    vocab.idx_to_token = json.load(open(os.path.join(data_dir, 'vocab.json')))
+    vocab.token_to_idx = {token: idx for idx, token in enumerate(
+        vocab.idx_to_token)}
+    bert = d2l.BERTModel(len(vocab), num_hiddens, norm_shape=[256],
+                        ffn_num_input=256, ffn_num_hiddens=ffn_num_hiddens,
```
Comment (astonzhang) [inline chapter_natural-language-processing-applications/natural-language-inference-bert.md:103]: indent after (

**Comment by mli** [comment]
Job d2l-en/PR-1554/1 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1554/

**Comment by mli** [comment]
Job d2l-en/PR-1554/4 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1554/

**Pull request merged**